File: PenetrationProjectAPI/src/main/java/com/ssafy/cafe/model/dao/OrderDetailDao.java
package com.ssafy.cafe.model.dao;

import java.util.List;

import com.ssafy.cafe.model.dto.OrderDetail;

public interface OrderDetailDao {

	int insert(OrderDetail detail);

	int delete(Integer detailId);

	List<OrderDetail> select(Integer orderId);

	List<OrderDetail> selectAll();

}

File: SmartStoreApp/app/src/main/java/com/ssafy/smartstore/database/OrderDetailDao.kt
package com.ssafy.smartstore.database

import androidx.room.*
import com.ssafy.smartstore.dto.OrderDetail
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface OrderDetailDao {

    @GET("rest/order/orderdetail/{orderId}")
    fun select(@Path("orderId")orderId: Int): Call<List<OrderDetail>>

    @POST("rest/order/orderdetail")
    fun insert(@Body body: OrderDetail): Call<Boolean>
}